db/migrations/20230207080938_categories.sql

```diff
@@ -0,0 +1,10 @@
+-- migrate:up
+CREATE TABLE product_categories (
+    id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    category_name VARCHAR(20),
+    product_id INT,
+    CONSTRAINT product_categories_product_id_fkey FOREIGN KEY (product_id) REFERENCES products (id)
+);
```

Comment:
- 제품을 포함하고 있는 최상단의 카테고리는 `main_categories`라는 테이블명으로 수정하는 것이 좋을 것 같습니다.
- `category_name`으로 컬럼을 짓게 되면 `main_categories.category_name` 과 같이 category라는 단어가 중복되기 때문에 `name`으로 지어도 좋을 것 같습니다.
- 테이블 간 관계를 파악하면, `main_categories` 테이블과 `products` 테이블은 1:N 관계입니다. 따라서 `main_categories` 테이블이 `products` 테이블을 참조하는 것이 아니라,그 반대가 되어야 합니다.

```suggestion
CREATE TABLE main_categories (
    id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
    name VARCHAR(20)
);
```